### PR TITLE
Tag data on maps from the python client.

### PR DESCRIPTION
**Pull request by AndriyMulyar**

### CODE REVIEW

File: nomic/project.py [hunk 1]
```diff
@@ -665,6 +665,86 @@ def _get_atoms(self, ids: List[str]) -> List[Dict]:
...
+        if response.status_code != 200:
+            raise Exception("Failed to add tags")
+
+    def delete_tags(self, datum_ids: List[str], tags: List[str], delete_all=False):
+        '''
+        Deletes the specified tags from the given datum_ids.
+        Args:
+            datum_ids: The datum_ids to delete tags from.
+            tags: The list of tags to delete from the data points.
+            delete_all: If true, ignores datum_ids and deletes all specified tags from all data points.
```
Comment (bmschmidt) [inline nomic/project.py:726]: Be clear that tags and datums_ids are for all.

```
            datum_ids: The datum_ids to delete tags from.
            tags: The list of tags to remove from all of the datum_ids.
            delete_all: If true, ignores datum_ids and deletes all specified tags from all data points.
```

File: nomic/project.py [hunk 2]
```diff
@@ -665,6 +665,86 @@ def _get_atoms(self, ids: List[str]) -> List[Dict]:
...
+        writer.close()
+        payload = buffer.getvalue()
+
+        headers = self.project.header.copy()
+        headers['Content-Type'] = 'application/octet-stream'
+        response = requests.post(self.project.atlas_api_path + "/v1/project/tag/add", headers=headers, data=payload)
+        if response.status_code != 200:
+            raise Exception("Failed to add tags")
+
+    def delete_tags(self, datum_ids: List[str], tags: List[str], delete_all=False):
```
Comment (bmschmidt) [inline nomic/project.py:720]: I would prefer `remove_tags` to `delete_tags`

File: nomic/project.py [hunk 3]
```diff
@@ -665,6 +665,86 @@ def _get_atoms(self, ids: List[str]) -> List[Dict]:
...
+        return label_to_datums
+
+    def tag(self, datum_ids: List[str], tags: List[str]):
+        '''
+
+        Args:
+            datum_ids: The datum ids you want to tag
+            tags: A list containing the tags you want to apply to these data points.
+
+        Returns:
```
Comment (bstadt) [inline nomic/project.py:699]: what does this return?

File: nomic/project.py [hunk 4]
```diff
@@ -665,6 +665,86 @@ def _get_atoms(self, ids: List[str]) -> List[Dict]:
         else:
             raise Exception(response.json())
 
+    def get_tags(self) -> Dict[str, str]:
```
Comment (bmschmidt) [inline nomic/project.py:668]: wrong type signature?: `Dict[str, List[str]]`?

Comment (bstadt) [inline nomic/project.py:668]: perhaps an option for passing a list of datum_ids whose tags you want would be useful? maybe not a blocker for this pr though

File: nomic/tests/test_atlas_client.py [hunk 5]
```diff
@@ -101,103 +101,111 @@ def test_map_embeddings():
...
-                   indexed_field='text',
-                   build_topic_model=True
-                   )
-
-    assert p.total_datums == 100
-    p.get_tags()
-    p.delete()
\ No newline at end of file
+#
+# def test_date_metadata():
```
Comment (bstadt) [inline nomic/tests/test_atlas_client.py:115]: ?

File: nomic/project.py [hunk 6]
```diff
@@ -665,6 +665,86 @@ def _get_atoms(self, ids: List[str]) -> List[Dict]:
...
+
+
+        '''
+        assert isinstance(datum_ids, list), 'datum_ids must be a list of strings'
+        assert isinstance(tags, list), 'tags must be a list of strings'
+
+        colname = json.dumps({'project_id': self.project.id, 'atlas_index_id': self.atlas_index_id, 'type': 'datum_id', 'tags': tags})
+        payload_table = pa.table([pa.array(datum_ids, type=pa.string())], [colname])
+        buffer = io.BytesIO()
+        writer = ipc.new_file(buffer, payload_table.schema, options=ipc.IpcWriteOptions(compression=None))
```
Comment (bmschmidt) [inline nomic/project.py:709]: Use `compression='zstd'`

File: nomic/project.py (hunk 4 above)
Comment (AndriyMulyar) [reply nomic/project.py]: Will keep just this for now. I want to see how users use it prior to writing different ways of getting tags.

**Review by bmschmidt — APPROVED** [review]

**Pull request merged**